float64 ratio
---
bool success   
string error_message 
